pkg/provider/onepassword/onepassword.go

```diff
@@ -299,6 +329,11 @@ func (provider *ProviderOnePassword) PushSecret(ctx context.Context, secret *cor
 		return ErrKeyNotFound
 	}
 
+	metadata, err := metadata.ParseMetadataParameters[PushSecretMetadataSpec](ref.GetMetadata())
+	if err != nil {
+		return fmt.Errorf("failed to parse push secret metadata: %w", err)
+	}
+
```

Comment:
Below this, in `createItem` the vault is also chosen. On line 224. I think you need to apply the metadata vault option there too.

Reply:
So I have been thinking..
Imho, changing the vault once it has already been created is undesirable, changing the vault would leave possible secret orphans in the vaults. I added an explanation in the docs https://github.com/external-secrets/external-secrets/pull/4173/files#diff-808f04183b4f2d32d1ee555aff0d60a1ec9a0c1cbf66ce0eec4ff0e508add8f5R218
In `createItem` selecting one of the vaults form the available ones makes sense to me

Reply:
Okay.